### DISCUSSION TITLE
Module Federation?

### DISCUSSION THREAD

**Discussion by iqtedar123** [oven-sh/bun#4653] [Q&A]
Will Bun support something like webpack federation?
Let's say I have multiple bun projects and I want to import them into a shell application.
Can it be done with bun? How?
Webpack uses module Federation to achieve this.
Thanks.

**Comment by RigottiG** [comment]
I'd also like to know.

**Comment by vitalics** [comment]
I suppose it works like [esbuild](https://esbuild.github.io/). It means if you have `cli` and `utils` package. And `cli` dependant on the `utils`. So Bun will build into 1 js file like esbuild does

**Comment by AllenAttuned** [comment]
I'm also interested. Module Federation is the main reason we're still using Webpack and hesitant to move to a different build tool.

[Webpack Module Federation docs](https://webpack.js.org/concepts/module-federation/)
